src/lib.rs

```diff
@@ -1,3 +1,75 @@
+//! Landlock is a security feature available since Linux 5.13.
+//! The goal is to enable to restrict ambient rights
+//! (e.g., global filesystem access)
+//! for a set of processes by creating safe security sandboxes as new security layers
+//! in addition to the existing system-wide access-controls.
+//! This kind of sandbox is expected to help mitigate the security impact of bugs,
+//! unexpected or malicious behaviors in applications.
+//! Landlock empowers any process, including unprivileged ones, to securely restrict themselves.
+//! More information about Landlock can be found in the [official website](https://landlock.io).
+//!
+//! This crate provides a safe abstraction for the Landlock system calls, along with some helpers.
+//!
+//! # Use cases
+//!
+//! This crate is especially useful for build-in application sandboxing:
+//! * Parsers hardening (e.g., archive tools, file format conversion, renderers).
+//! * (Part of) applications with limited file renaming or linking needs
+//!   (e.g., some system or network services).
+//! * Monolith applications dealing with different levels of confidentiality
+//!   (e.g., web browser, email server).
+//!
+//! # Examples
+//!
+//! Examples can be found with the [`Ruleset` documentation](Ruleset)
+//! and the [sandboxer example](https://github.com/landlock-lsm/rust-landlock/blob/master/examples/sandboxer.rs).
+//!
+//! # Current limitations
+//!
+//! This crate handles the Landlock features available since Linux 5.13
+//! and then inherits some [kernel limitations](https://www.kernel.org/doc/html/latest/userspace-api/landlock.html#current-limitations)
```

Comment:
(fyi) Good idea to spell this out in the documentation, I'll copy the warning about these two items into the go-landlock documentation as well.